src/bridge/cockpitdbusjson.c

```diff
@@ -1621,6 +1621,53 @@ on_cache_meta (CockpitDBusCache *cache,
   json_object_unref (message);
 }
 
+static void
+handle_dbus_meta (CockpitDBusJson *self,
+                  JsonObject *object)
+{
+  CockpitChannel *channel = COCKPIT_CHANNEL (self);
+  GDBusInterfaceInfo *iface;
+  JsonObject *interface;
+  GError *error = NULL;
+  JsonObject *meta;
+  GList *names, *l;
+  JsonNode *node;
+
+  node = json_object_get_member (object, "meta");
+  g_return_if_fail (node != NULL);
+
+  if (!JSON_NODE_HOLDS_OBJECT (node))
+    {
+      cockpit_channel_fail (channel, "protocol-error", "incorrect \"meta\" field in dbus command");
+      return;
+    }
+
+  meta = json_node_get_object (node);
+  names = json_object_get_members (meta);
+  for (l = names; l != NULL; l = g_list_next (l))
+    {
+      if (!cockpit_json_get_object (meta, l->data, NULL, &interface))
+        {
+          cockpit_channel_fail (channel, "protocol-error", "invalid interface in dbus \"meta\" command");
+          break;
+        }
+
+      iface = cockpit_dbus_meta_parse (l->data, interface, &error);
+      if (iface)
+        {
+          g_hash_table_insert (self->interface_info, iface->name, iface);
+        }
+      else
+        {
+          cockpit_channel_fail (channel, "protocol-error", "%s", error->message);
+          g_error_free (error);
+          break;
```

Comment:
We close the entire channel. We don't keep anything. This is a programmer error.